morl_baselines/multi_policy/pgmorl/pgmorl.py

```diff
@@ -615,7 +615,7 @@ def train(self):
         # Evolution
         remaining_iterations = max(self.max_iterations - self.warmup_iterations, self.evolutionary_iterations)
```

Comment:
@mathieu-reymond what do you think about fixing that line instead?

Reply:
changed it, what do you think?

Reply:
It feels strange to me that the algorithm can run for more than `max_iterations` tho. Maybe we should rename those variables? Why do we need `max_iterations`? Can't the user provide only warmup and evolutionary iterations?

Reply:
This is due to the way PGMORL sets the number of samples. On the one hand, it specifies a fixed number of timesteps. On the other hand, it asks to do `warmup` + `evolutionary` iterations of `steps_per_iteration`. The whole logic is very weird but that's how it is described in the paper. 
<img width="946" alt="image" src="https://user-images.githubusercontent.com/11799929/216561190-2a2cc6a6-27e7-4a5f-9cf7-01248f12231d.png">

I scratched my head for some time to understand what they meant. And I saw the original implementation: https://github.com/mit-gfx/PGMORL/blob/master/morl/morl.py#L40

If you have suggestions to make the code easier I'm all for it.

Reply:
Ah, I thought max_iteration was a parameter of the algorithm, and not a internal variable. In that case, I think the current change is good